### PR TITLE
IP APIDetailsStorage

### PR DESCRIPTION
**Pull request by skorpiom**
Hey,

Some logic responsible for handling api key tile. 

Sorry for the mess with files - in this pull req important are just named 'APIDetailsStorage'. 

If you want to start it:

`npm run start:apiKeyLogic`

### CODE REVIEW

File: Src/Logic/APIDetailsStorage/APIDetailsStorage.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { DetailsAPI } from "../../../Models/DetailsAPI.model";
+import { saveInLocalStorage, readFromLocalStorage } from "../../../Src/Logic/LocalStorage/LocalStorage"
+
+const NO_USER_ERROR = `User doesn't exist!`;
+const NO_API_DETAILS_ERROR = `User doesn't contain API details.`;
+const SAVED_CORRECTLY_CONFIRM = 'API DETAILS saved correctly!';
```
Comment (robert-czwartosz) [inline Src/Logic/APIDetailsStorage/APIDetailsStorage.ts:6]: I suggest to move these lines to new consts.ts file.
And import const.ts to this file and test.ts file.

Comment (skorpiom) [reply Src/Logic/APIDetailsStorage/APIDetailsStorage.ts:6]: That's great suggestion but I am not sure in terms of project architecture where I should place that file.
New folder 'Constants' in main directory and over there file like LocalStorage.ts?
Or just 'const.ts' in ./Src/ ?

**Review by fszwedo — APPROVED** [review]
works and looks great, apart from one small suggestion I left in the UI component pull request all good from me

File: Src/Logic/APIDetailsStorage/APIDetailsStorage.ts (hunk 1 above)
Comment (robert-czwartosz) [reply Src/Logic/APIDetailsStorage/APIDetailsStorage.ts:6]: In JS project contants files were added to file next to component file.
I think this question should be adressed to Tech Lead @JustynaSmoliga

File: Test/APIDetailsStorage.test.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    setItem(key, value) {
+        this.store[key] = String(value);
+    }
+
+    removeItem(key) {
+        delete this.store[key];
+    }
+};
+
+global["localStorage"] = new LocalStorageMock;
```
[the remark this answers is not in the document]
Comment (skorpiom) [reply Test/APIDetailsStorage.test.ts]: You're right - deleted.

File: Src/Logic/APIDetailsStorage/APIDetailsStorage.ts (hunk 1 above)
Comment (JustynaSmoliga) [reply Src/Logic/APIDetailsStorage/APIDetailsStorage.ts:6]: I think, that we should chose option "New folder 'Constants' in main directory and over there file like const.ts".

**Comment by skorpiom** [comment]
I created in root dir folder Contants and placed there consts.ts file.

**Review by robert-czwartosz — APPROVED** [review]

**Pull request merged**